test/e2e/scrapeconfig_test.go

```diff
@@ -491,559 +491,578 @@ func testScrapeConfigDNSSDConfig(t *testing.T) {
 	require.NoError(t, err)
 }
 
+type scrapeCRDTestCase struct {
+	name             string
+	scrapeConfigSpec monitoringv1alpha1.ScrapeConfigSpec
+	expectedError    bool
+}
+
 func testScrapeConfigCRDValidations(t *testing.T) {
 	t.Parallel()
-	name := "test"
+	t.Run("KubernetesSD_API_Validations", func(t *testing.T) {
+		runScrapeConfigCRDValidationTCS(t, K8STestCases)
+	})
+	t.Run("DNSSD_API_Validations", func(t *testing.T) {
+		runScrapeConfigCRDValidationTCS(t, DNSSDTestCases)
+	})
+	t.Run("EC2SD_API_Validations", func(t *testing.T) {
+		runScrapeConfigCRDValidationTCS(t, EC2SDTestCases)
+	})
+}
 
-	tests := []struct {
-		name             string
-		scrapeConfigSpec monitoringv1alpha1.ScrapeConfigSpec
-		expectedError    bool
-	}{
-		{
-			name: "KubernetesSDConfigs: APIServer with empty value",
-			scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
-				KubernetesSDConfigs: []monitoringv1alpha1.KubernetesSDConfig{
-					{
-						Role:      "EndpointSlice",
-						APIServer: ptr.To(""),
-					},
+func runScrapeConfigCRDValidationTCS(t *testing.T, testCases []scrapeCRDTestCase) {
```

Comment:
TCS is for TestCase?
```suggestion
func runScrapeConfigCRDValidation(t *testing.T, testCases []scrapeCRDTestCase) {
```